app/controllers/application_controller.rb

```diff
@@ -3,13 +3,18 @@ class ApplicationController < ActionController::Base
 
   helper_method :current_user_id
   helper_method :ensure_user_signed_in
+  helper_method :current_user
 
   private
 
   def current_user_id
     session[:auth_id]
   end
 
+  def current_user
+    @current_user = API::User.find(auth_id: current_user_id).first
```

Comment:
I think you want `@current_user ||= API::User.find(auth_id: current_user_id).first` here. That would be caching the response to an instance variable.

Reply:
So if @current_user is nil assign it otherwise use the existing value?

Reply:
Exactly

Reply:
So nice, thanks!